streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -237,7 +237,14 @@ public void setUncaughtExceptionHandler(Thread.UncaughtExceptionHandler eh) {
      * @return  A facade wrapping the {@link org.apache.kafka.streams.processor.StateStore} instances
      */
     public <T> T getStore(final String storeName, final QueryableStoreType<T> queryableStoreType) {
+        validateIsRunning();
         return queryableStoreProvider.getStore(storeName, queryableStoreType);
     }
 
+    private void validateIsRunning() {
+        if (state != RUNNING) {
```

Comment:
The only problem it will cause is that you might not find the store if you called `stores(...)` immediately after `start`.